### DISCUSSION TITLE
Unable to redo migrations

### DISCUSSION THREAD

**Discussion by me-ZaidAli** [diesel-rs/diesel#3992] [Q&A] [answered]
Hi
I have been trying to redo a couple of migrations but getting this error
`Unable to find migration version 20240408094742 to revert in the migrations directory.`

Now from this [discussion](https://github.com/diesel-rs/diesel/discussions/2719), I figured there must be a migration in `__diesel_schema_migrations` which isn't in migrations directory anymore but I can't figure out how to remove the migration from `__diesel_schema_migrations`. Any help would be much appreciated.

**Comment by weiznich** [comment, marked as answer]
We do not provide an API for removing interacting with that internal table without having the required migrations available. Technically you could use your database CLI tool to manually delete the corresponding entry, but at that point you are far outside of the supported functionality.

**Reply by me-ZaidAli** [reply]
Makes sense. I just deleted the migration entry from the `__diesel_schema_migrations` in db and it worked. Thanks for the help.
